openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -333,7 +333,7 @@ This endpoint is used by a Credential Issuer that is already interacting with an
 
 ## Credential Offer {#credential-offer}
 
-The Credential Issuer sends Credential Offer using an HTTP GET request or an HTTP redirect to the Wallet's Credential Offer Endpoint defined in (#client-metadata).
+The Credential Issuer makes a Credential Offer by allowing the End-User to invoke the Wallet using the Wallet's Credential Offer Endpoint defined in (#client-metadata) (for example by clicking a link) and/or rendering a QR code containing the Credential Offer that the End-User can scan in a wallet or an arbitrary camera application.
```

Comment:
I am unsure exactly what use case you are describing, but either my guess would be that either:

1. This is happening in the user's browser/on the user's device and hence I'm pretty sure a click is involved so this text is fine, or:
2. This is not involving the user's browser/device and is already impossible to do in VCI today without custom extensions

Is this flow you're talking about one you've implemented - if so can you provide any extra details on how it works?

Reply:
Thanks! So in this case the wallet's `credential_offer_endpoint` is specific to the user rather than generic to the wallet software? Or if not how does it know what end user the offer is for?

Reply:
Yes, in this particular case every user has a specific wallet that has a specific URL that is tied to it.

Reply:
Okay, thanks. I think I need to think about this...

When the issuer does the http GET on the `credential_offer_endpoint` what is the wallet's HTTP response in your case?

Reply:
202 Accepted

Reply:
I was wondering about that case as well when reviewing this PR. For a backend-based wallet, the Wallet wouldn't be invoked, but there would just be a direct call to the `credential_offer_endpoint`. Would it make sense to relax the language a bit here or add a special case if the Credential Offer Endpoint is a callable (https) URL?

Reply:
@nemqe A further clarify question if you don't mind - in this case do the wallets all have the same client_id, or is there a separate client_id for each user's wallet?

Reply:
And actually one more question:

How does the issuer know if it should make a backend call or send the user to the url?

Reply:
We originally implemented the `pre-auth` code flow with anonymous access (no client ids).

As for the how does an issuer know what to do, well we kind of made some custom logic before the openid protocol is triggered that sets up things out of band, so the issuer never really dynamically decides if it is going to show a link, or a qr code, the assumption is that once the business specific logic is performed the issuance always starts by just sending an offer to the wallet.

Reply:
I think in the early days we tried to advocate for some of these client metadata concepts but in the end as we did not have too much luck we resorted to creating our own `openid-credential-wallet` file with the endpoint inside it as I think the consensus then was "wallet endpoint discovery is out of the scope of the spec".

My comment here was just about the language as in the current spec text we are not out of compliance, but it seemed to me that the new text from this PR concentrates fully on the Mobile Wallets and Web Browsers and the involvement of the User in the Wallet invocation so the concern was that we will fall out of compliance as a consequence.

Reply:
@nemqe Would a statement like this solve your concern:

> Issuers MAY also communicate credential offers directly to a Wallet's backend but any mechanism for doing so is currently outside of the scope of this specification.

?

Reply:
Yes, I think that would suffice. Thank you.

Reply:
Okay, great, I've added that now, thanks!